src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,343 @@
+        {
+            if (_options.ConfigurationRefresherEndPointUri == null && _options.DynamicConfigs == null)
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+                return;
+            }
+            await _options.ConfigurationRefresherEndPointUri.Refresh();
+            bool.TryParse(_options.DynamicConfigs["FeatureManagement:DependencyUpdateErrorProcessor"],
+                out var dependencyUpdateErrorProcessorFlag);
+            if (dependencyUpdateErrorProcessorFlag)
+            {
+                IReliableDictionary<string, DateTimeOffset> update =
+                    await _stateManager.GetOrAddAsync<IReliableDictionary<string, DateTimeOffset>>("update");
+                try
+                {
+                    DateTimeOffset previousTransaction;
+                    using (ITransaction tx = _stateManager.CreateTransaction())
+                    {
+                        previousTransaction = await update.GetOrAddAsync(
+                         tx,
+                         "update",
+                         //DateTimeOffset.UtcNow 
+                         new DateTime(2002, 10, 18)
+                         );
+                        await tx.CommitAsync();
+                    }
+                    await CheckForErrorInRepositoryBranchHistoryTable(previousTransaction, _options.GithubUrl, update);
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError(ex.Message, "Unable to get the previous transaction time from reliable services.");
+                }
+            }
+        }
+
+        private async Task CheckForErrorInRepositoryBranchHistoryTable(DateTimeOffset previousTransaction , string issueRepo , IReliableDictionary<string, DateTimeOffset> update)
+        {
+            try
+            {
+                // Looking for errors from the RepositoryBranchHistory table
+                List<RepositoryBranchUpdateHistoryEntry> unprocessedHistoryEntries = (from repoBranchUpdateHistory in Context.RepositoryBranchUpdateHistory
+                    where repoBranchUpdateHistory.Success == false
+                    where repoBranchUpdateHistory.Timestamp > previousTransaction.UtcDateTime
+                    orderby repoBranchUpdateHistory.Timestamp ascending
+                    select repoBranchUpdateHistory).ToList();
+                if (!unprocessedHistoryEntries.Any())
+                {
+                    Logger.LogInformation("No errors found in the RepositoryBranchUpdateHistory table.");
+                    return;
+                }
+                Logger.LogInformation("Going to create the github issue.");
+
+                foreach (var error in unprocessedHistoryEntries)
+                {
+                    try
+                    {
+                        await CreateOrUpdateGithubIssueAsync(error, issueRepo);
```

Comment:
I think there's several advantages to moving the switch statement that decides what to do with each error type here similar to my suggestion in [the other pr](https://github.com/riarenas/arcade-services/pull/1#discussion_r387854858)

* You can still build the description of the error depending on the method.
* You can create a Func<string, bool> object that represents how you search for the issue text inside a given comment or the issue body.
```C#
Func<string, bool> errorForUpdateAssetsAsync = (body) =>
{
  Dictionary<string, string> argumentParser = parseIssueCommentBody(body);
  string subId = GetSubscriptionId(repositoryBranchUpdateHistory.Arguments);
  if (string.Equals(argumentParser["subscriptionId"], subId) &&
      string.Equals(argumentParser["errorMessage"], repositoryBranchUpdateHistory.ErrorMessage))
  {
    return true;
  }
  return false;
}
```
* CreateOrUpdateGitHubIssueAsync now becomes something like
```C#
CreateOrUpdateGithubIssueAsync(RepositoryBranchHistoryUpdateEntry error,
  string descriptionToWrite,
  Func<string,bool> isErrorMessageInDescription,
  string issueRepo)
{
   <get issue body>
   if (isErrorMessageInDescription(issueBody))
   {
     <update the issue body>
     flag = true
   }
   if (!flag)
   {
     foreach (comment in issue)
     {
       if (isErrorMessageInDescription(commentBody)
       {
        <update comment body>
         flag = true
         break;
       }
    }
   }
   if (!flag)
   {
     <write error in new comment>
   }
   
  }  
}
```

This way, you guard future devs from doing the wrong thing, as they know they need to pass all the parameters to `CreateOrUpdateGithubissueAsync`, and adding a new method that should write errors is reduced to two things:

* building the description
* building the func that will search for the error in the body or comment.

You already have most of the code for this, it's just a matter of moving things around so that there's only a single point where you can miss things.

Let me know if that all makes sense.

Reply:
In the switch statement I added both 

1) check if the issue has to be updated or not 
2) description for the issue / comment body 

Please take a look,